tasks/fr/fr_income.py

```diff
@@ -0,0 +1,197 @@
+from collections import OrderedDict
+from luigi import IntParameter, Parameter, WrapperTask, Task, LocalTarget
+import os
+import glob
+from tasks.meta import OBSTable, OBSColumn, OBSTag
+from tasks.tags import SectionTags, SubsectionTags, UnitTags
+from tasks.fr.insee import OutputAreaColumns
+import csv
+import pandas as pd
+
+
+class DownloadFRIncomeIris(Task):
+
+    table_theme = Parameter()
+
+    URL_base = 'https://www.insee.fr/fr/statistiques/fichier/2507751/'
+
+    def download(self):
+
+        themes = {
+            'BASE_TD_FILO_DEC_IRIS_2012': 'BASE_TD_FILO_DEC_IRIS_2012.xls',
+            'BASE_TD_FILO_DISP_IRIS_2012': 'BASE_TD_FILO_DISP_IRIS_2012.xls',
+        }
+
+        URL = self.URL_base + themes.get(self.table_theme)
+
+        shell('wget -P {output} {url}'.format(
+           output=self.output().path,
+           url=URL
+        ))
+
+    def run(self):
+        self.output().makedirs()
+        self.download()
+
+    def output(self):
+        return LocalTarget(os.path.join('tmp', classpath(self), self.task_id))
+
+
+class RawIncomeIrisData(CSV2TempTableTask):
+
+    table_theme = Parameter()
+
+    def requires(self):
+        return DownloadFRIncomeIris(table_theme=self.table_theme)
+
+    def xls2csv(self):
+        # Read in excel file, it should be the only file in self.input().path
+        # xls = pd.ExcelFile(os.path.join(self.input().path, os.listdir(self.input().path)[0]))
+        xls = pd.ExcelFile(glob.glob(os.path.join(self.input().path,
+                                                  '{table_theme}.xls'.format(table_theme=self.table_theme)))[0])
+
+        # Remove header
+        df = xls.parse(skiprows=5, header=0)
+        df.to_csv(os.path.join(self.input().path, '{table_theme}.csv'.format(table_theme=self.table_theme)),
+                  index=False, encoding='utf8')
+
+    def input_csv(self):
+        if not glob.glob(os.path.join(self.input().path, '{table_theme}.csv'.format(table_theme=self.table_theme))):
+            self.xls2csv()
+
+        return os.path.join(self.input().path, '{table_theme}.csv'.format(table_theme=self.table_theme))
+
+
+class SourceTags(TagsTask):
+    def version(self):
+        return 1
+
+    def tags(self):
+        return [
+            OBSTag(id='insee',
+                   name='INSEE',
+                   type='source',
+                   description='http://www.insee.fr/fr/bases-de-donnees/default.asp?page=recensement/resultats/2012/donnees-detaillees-recensement-2012.htm')
+        ]
+
+
+class IrisIncomeColumns(ColumnsTask):
+
+    table_theme = Parameter()
+
+    def requires(self):
+        return {
+            'sections': SectionTags(),
+            'subsections': SubsectionTags(),
+            'unittags': UnitTags(),
+            'sourcetag': SourceTags()
+        }
+
+
+    def version(self):
+        return 1
+
+    def columns(self):
+        cols = OrderedDict()
+        input_ = self.input()
```

Comment:
This is related to an unmerged change PR #150 , if we're going to do it this way we should merge that PR